redisinsight/api/src/modules/database-recommendation/providers/database-recommendation.provider.ts

```diff
@@ -141,6 +150,30 @@ export class DatabaseRecommendationProvider {
     return model;
   }
 
+  /**
+   * Sync db analysis recommendations with live recommendations
+   * @param clientMetadata
+   * @param dbAnalysisRecommendations
+   * @param liveRecommendations
+   */
+  async sync(
+    clientMetadata: ClientMetadata,
+    dbAnalysisRecommendations: Recommendation[],
+    liveRecommendations: DatabaseRecommendation[],
+  ): Promise<void> {
+    this.logger.log('Synchronization of recommendations');
+    try {
+      const newRecommendations = differenceBy(dbAnalysisRecommendations, liveRecommendations, 'name');
```

Comment:
Fine with me for now but I feel like previously we wanted to sync recommendations one-by-one.
To have generic mechanism to create new and ignore existing recommendations. the same way as we are determining recommendations. 
With this approach you might have several the same recommendations with concurrent sync invocations, no?
Do you have unique constraint on db level?

Reply:
Good point